streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -225,6 +229,30 @@ protected int getVersionThreeByteLength(final byte[] endPointBytes) {
                4 + standbyTasks.size() * 8 + // length + standby tasks
                4 + endPointBytes.length; // length + userEndPoint
     }
+    
+    private ByteBuffer encodeVersionFour() {
```

Comment:
more of a meta comment but will we have to add `encodeVersionX` for each new version?  

Just off the top of my head is maybe we could have more generic methods taking a version number parameter and within the method, we use a `switch` statement to take the version specific action.   I don't think we need to take this on now though.